cernlib: build failure on upcoming gcc-11 and binutils-2.37. Consider removing?

### Describe the bug

`cernlib` (https://cernlib.web.cern.ch/cernlib/) is an old unmaintained software from 2006.

Currently fails to build on `gfortran-11` due to `fortran` typing changes:

```
cernlib> /build/2006/src/packlib/cspack/xz/xzctof.F:40:27:
cernlib>    40 |       DATA         IPATT/X'0123CDEF'/
cernlib>       |                           1
cernlib> Error: Hexadecimal constant at (1) uses nonstandard X instead of Z [see '-fno-allow-invalid-boz']
cernlib> make[3]: *** [Makefile:221: archive/xzctof.o] Error 1
```

And also on `binutils-2.37` that will need an `ar l` fix: https://github.com/NixOS/nixpkgs/pull/134917

WDYT of removing `cernlib` instead of patching it? Is it still useful?

CC @veprbl 

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: cernlib
```

--- a/pkgs/development/libraries/physics/cernlib/default.nix
+++ b/pkgs/development/libraries/physics/cernlib/default.nix
@@ -35,18 +35,30 @@ stdenv.mkDerivation rec {
     export PATH=$PATH:$CERN_ROOT/bin
   '';
 
+  FFLAGS = lib.optionals (lib.versionAtLeast gfortran.version "10.0.0") [
+    # Fix https://github.com/vmc-project/geant3/issues/17
+    "-fallow-invalid-boz"
+
+    # Fix for gfortran 10
+    "-fallow-argument-mismatch"
+  ];
+
+  makeFlags = [
+    "FORTRANOPTIONS=$(FFLAGS)"
+  ];
+
   buildPhase = ''
     cd $CERN_ROOT
     mkdir -p build bin lib
 
     cd $CERN_ROOT/build
     $CVSCOSRC/config/imake_boot
-    make -j $NIX_BUILD_CORES bin/kuipc
-    make -j $NIX_BUILD_CORES scripts/Makefile
+    make -j $NIX_BUILD_CORES $makeFlags bin/kuipc
+    make -j $NIX_BUILD_CORES $makeFlags scripts/Makefile
     pushd scripts
-    make -j $NIX_BUILD_CORES install.bin
+    make -j $NIX_BUILD_CORES $makeFlags install.bin
     popd
-    make -j $NIX_BUILD_CORES
+    make -j $NIX_BUILD_CORES $makeFlags
   '';
 
   installPhase = ''

